.github/workflows/CI.yml

```
@@ -61,12 +61,65 @@ jobs:
       - name: "Run tests"
         uses: julia-actions/julia-runtest@latest
         with:
-          coverage: ${{ matrix.julia-version == '1.6' }}
+          annotate: ${{ matrix.julia-version == '1.8' }}
+          coverage: ${{ matrix.julia-version == '1.8' }}
           depwarn: error
       - name: "Process code coverage"
+        if: matrix.julia-version == '1.8'
         uses: julia-actions/julia-processcoverage@v1
         with:
           directories: src,experimental
       - name: "Upload coverage data to Codecov"
+        if: matrix.julia-version == '1.8'
+        continue-on-error: true
+        uses: codecov/codecov-action@v3
+
+  doctest:
+    runs-on: ${{ matrix.os }}
+    timeout-minutes: 150
+    continue-on-error: ${{ matrix.julia-version == 'nightly' }}
+    strategy:
+      fail-fast: false
+      matrix:
+        julia-version:
+          - '1.7'
+          - '1.8'
+          - '~1.9.0-0'
+          - 'nightly'
+        os:
+          - ubuntu-latest
```

```suggestion
          - ubuntu-latest
        include:
          - julia-version: '1.8'
            os: macOS-latest
```
I would prefer to keep at least one macos doctest job, I think there is quite a bit of code that is only covered by doctests. Would this require anything other than this `include` section (and another of the very few workers ...)?

---

I think this patch is sufficient, we'll try. 

But using 50% more macOS workers is painful; I've had multiple occasions in the past weeks were macOS runners were still playing catch up with our various PRs and master updated in the evening after I was already at home :-(.

Originally I tried to do what we discussed above and the macOS runners just run both regular tests and doctests. This worked but I felt the result was simply to hard to understand, debug, and modify. So now I've split the job matrices for the two, and I think the result is quite a bit better, esp. since the job name now makes it clear whether regular tests or doctests are run.

Anyway, in this new setup, it wasn't too bad to make it a bit more ugly again and modify the new macOS doctes runner to also run regular tests... and then drop the "regular test runner" for mac.

Let's see if that works (it rather late, so I wouldn't be surprised if I screwed it up again *sigh*)